geode-assembly/src/distributedTest/java/org/apache/geode/rest/internal/web/controllers/RestAPIsWithSSLDUnitTest.java

```diff
@@ -695,10 +412,36 @@ public void testSSLWithMultipleCipherSuite() throws Exception {
     ssl.init(null, null, new java.security.SecureRandom());
     String[] cipherSuites = ssl.getSocketFactory().getSupportedCipherSuites();
 
-    props.setProperty(SSL_CIPHERS, cipherSuites[0] + "," + cipherSuites[1]);
+    // This is the safest in terms of support across various JDK releases
+    String rsaCiphers =
+        Arrays.stream(cipherSuites).filter(c -> c.contains("RSA")).collect(Collectors.joining(","));
+    props.setProperty(SSL_CIPHERS, rsaCiphers);
+
+    startClusterWithSSL(props);
+    validateConnection(props);
+  }
+
+  @Test
+  public void testSSLWithMultipleCipherSuiteLegacy() throws Exception {
+    System.setProperty("javax.net.debug", "ssl,handshake");
+    Properties props = new Properties();
+    props.setProperty(HTTP_SERVICE_SSL_ENABLED, "true");
+    props.setProperty(HTTP_SERVICE_SSL_KEYSTORE,
+        findTrustedJKSWithSingleEntry().getCanonicalPath());
+    props.setProperty(HTTP_SERVICE_SSL_KEYSTORE_PASSWORD, "password");
+    props.setProperty(HTTP_SERVICE_SSL_PROTOCOLS, "TLSv1.2");
+
+    SSLContext ssl = SSLContext.getInstance("TLSv1.2");
+
+    ssl.init(null, null, new java.security.SecureRandom());
+    String[] cipherSuites = ssl.getSocketFactory().getSupportedCipherSuites();
+
+    String rsaCiphers =
+        Arrays.stream(cipherSuites).filter(c -> c.contains("RSA")).collect(Collectors.joining(","));
+    props.setProperty(HTTP_SERVICE_SSL_CIPHERS, rsaCiphers);
```

Comment:
Is it worth extracting getting and filtering the cipher suites as a method. It looks like the code for this is duplicated (except for the property name) for this and the preceding test.